include/common/Driver.hpp

```diff
@@ -458,6 +467,24 @@ class Driver
 		else
 			_writeLastState = false;
 
+		// Adding a new output variable
+		std::ostringstream oss;
+		for (int i = 0; i <= _sim->model()->maxUnitOperationId(); ++i)
+		{
+			oss.str("");
+			oss << "unit_" << std::setfill('0') << std::setw(3) << std::setprecision(0) << i;
+			pp.pushScope(oss.str());
+			if (pp.exists("WRITE_SOLUTION_LAST_UNIT"))
+			{
+				_writeLastStateUnit = pp.getBool("WRITE_SOLUTION_LAST_UNIT");
+				_flag[i] = true;
+			}
+				
+			else
+				_writeLastStateUnit = false;
```

Comment:
In the new approach (see below). We only need to add the unit operation id of the unit to the list / vector if the `WRITE_SOLUTION_LAST_UNIT` is true.